c++/src/Int128.cc

```diff
@@ -488,4 +488,84 @@ namespace orc {
     return value;
   }
 
+  std::pair<bool, Int128> convertDecimal(Int128 value, int32_t fromScale, int32_t toPrecision,
+                                         int32_t toScale, bool round) {
+    std::pair<bool, Int128> result;
+    bool negative = value < 0;
+    result.second = value.abs();
+    result.first = false;
+
+    Int128 upperBound = scaleUpInt128ByPowerOfTen(1, toPrecision, result.first);
```

Comment:
nit: check result.first in case of invalid toPrecision

Reply:
Or assert fromScale, toPrecision, toScale are in the valid ranges.

Reply:
> Or assert fromScale, toPrecision, toScale are in the valid ranges.

done